### DISCUSSION TITLE
Throwing an error ends up as Uncaught Exception in VS Code

### DISCUSSION THREAD

**Discussion by johanlantz** [rrousselGit/riverpod#2467] [Q&A]
I might be doing this wrong but I I have FutureProvider that reads from a local database and I want to be able to handle errors by throwing and then catching them in the widget using riverpod, something like:
```
@riverpod
Future<Report> getActivity(
    GetActivityRef ref, DateTime date) async {
 
  final activityData = await getReportForDate(date);
  if (activityData.hasError()) {
    throw Exception();
  } else {
    return activityData.report;
  }
}
```

and in the widget:
```
when(
              data: (activityReport) => Text('OK),
              error: (error, _) {
                debugPrint(error.toString());
                return const Text('Error')
              },
              loading: () => Text('Loading'),
            ),
```

This works, if I throw in the repository code, it ends up in the error block above. What confuses me is that if I have "Uncaught Exceptions" selected in VS Code, then VS Code catches this exception instead and breaks so it never gets to the riverpod error handler.

Am I missing some step here to make VS Code happy and not treat this as an uncaught exception?

**Comment by rrousselGit** [comment]
Are you in an older Riverpod version? This shouldn't be a thing anything since 2.3.3

**Reply by johanlantz** [reply]
Thanks @rrousselGit for the super fast reply.

I was on 2.3.2 but now I have updated to:
hooks_riverpod: ^2.3.4
riverpod_annotation: ^2.0.4

and 
riverpod_generator: ^2.1.6
riverpod_lint: ^1.2.0

I am on VS Code 1.77.3 on Mac (latest)

I have done flutter pub get and even deleted the 2.3.2 version from the cache so I only have hooks_riverpod 2.3.4 now. 

I have also done flutter clean, restarted VS Code, and re-generated the .g file but the issue still remains. If I have Uncaught Exceptions selected, the throw is caught by VS Code.

Is there anything else I can check to help isolate the issue?

**Reply by johanlantz** [reply]
I also tried putting a breakpoint here: https://github.com/rrousselGit/riverpod/blob/master/packages/riverpod/lib/src/framework/element.dart#L427

but it does not trigger.
